pkg/mesh-networking/reconciliation/errhandler.go

```
@@ -0,0 +1,98 @@
+package reconciliation
+
+import (
+	"encoding/json"
+
+	discoveryv1alpha2 "github.com/solo-io/gloo-mesh/pkg/api/discovery.mesh.gloo.solo.io/v1alpha2"
+	"github.com/solo-io/gloo-mesh/pkg/api/networking.mesh.gloo.solo.io/input"
+	networkingv1alpha2 "github.com/solo-io/gloo-mesh/pkg/api/networking.mesh.gloo.solo.io/v1alpha2"
+	"github.com/solo-io/gloo-mesh/pkg/mesh-networking/translation/utils/metautils"
+	"github.com/solo-io/skv2/contrib/pkg/output"
+	v1 "github.com/solo-io/skv2/pkg/api/core.skv2.solo.io/v1"
+	"github.com/solo-io/skv2/pkg/ezkube"
+)
+
+var _ output.ErrorHandler = errHandler{}
+
+type errHandler struct {
+	inp input.Snapshot
+}
+
+func (e errHandler) HandleWriteError(resource ezkube.Object, err error) {
+	annotations := resource.GetAnnotations()
+	parentsStr, ok := annotations[metautils.ParentLabelkey]
+	if !ok {
+		return
+	}
+	var allParents map[string][]*v1.ObjectRef
+	if err := json.Unmarshal([]byte(parentsStr), &allParents); err != nil {
+		return
```

Can we surface this error somehow, maybe through logging?

---

The logging I've seen invoked uses `contextutils.LoggerFrom(ctx).Etc`, but there is no context here, should I pass it in, or are there other logging options?

I would also like to use this to handle the errors in `AppendParent` so that way that function does not need to return an error.